draft-mouris-cfrg-mastic.md

```diff
@@ -216,11 +217,103 @@ into Mastic in {{vidpf}}. The appendix includes supplementary material:
+| `xor`             | Function  | {{Section 2 of !VDAF}}|
+| `zeros`           | Function  | {{Section 2 of !VDAF}}|
+{: #common-functions title="Common Functionalities from {{!VDAF}}."}
+
+Mastic also uses finite fields as specified in {{Section 6.1 of !VDAF}}. We
+usually denote a finite field by `F` and its Python class object, a subclass of
+`Field`, as `field: type[F]`.
+The following functionalities are as defined in {{Section 6.1 of !VDAF}}:
+
+| Functionality       | Type          | Definition                |
+|:--------------------|:--------------|:--------------------------|
+| `Field`             | Constructor   | {{Section 6.1 of !VDAF}}  |
+| `field.encode_vec`  | Class Method  | {{Section 6.1.1 of !VDAF}}|
+| `field.zeros`       | Class Method  | {{Section 6.1 of !VDAF}}  |
+| `vec_add`           | Function      | {{Section 6.1.1 of !VDAF}}|
+| `vec_neg`           | Function      | {{Section 6.1.1 of !VDAF}}|
+| `vec_sub`           | Function      | {{Section 6.1.1 of !VDAF}}|
+{: #field-functionalities title="Finite Field Functionalities from {{!VDAF}}."}
+
+
+Moreover, the following algorithms of Fully Linear Proofs (FLPs) are as
+defined in {{Section 7.1 of !VDAF}}:
+
+| Functionality   | Type          | Definition                  |
+|:----------------|:--------------|:----------------------------|
+| `flp.decide`    | Class Method  | {{Section 7.1 of !VDAF}}    |
+| `flp.decode`    | Class Method  | {{Section 7.1.1 of !VDAF}}  |
+| `flp.encode`    | Class Method  | {{Section 7.1.1 of !VDAF}}  |
+| `flp.prove`     | Class Method  | {{Section 7.1 of !VDAF}}    |
+| `flp.query`     | Class Method  | {{Section 7.1 of !VDAF}}    |
+| `flp.truncate`  | Class Method  | {{Section 7.1.1 of !VDAF}}  |
+{: #FLP-functionalities title="FLP Functionalities from {{!VDAF}}."}
+
+The following parameters are defined in {{Section 7.3.2 of !VDAF}}:
+
+| Parameter     | Type      | Definition                  |
+|:--------------|:----------|:----------------------------|
+| `MEAS_LEN`    | integer   | {{Section 7.3.2 of !VDAF}}  |
+| `OUTPUT_LEN`  | integer   | {{Section 7.3.2 of !VDAF}}  |
+{: #circ-params title="Validity Circuit Parameters."}
+
+Mastic also uses eXtendable Output Functions (XOFs) as specified in
+{{Section 6.2 of !VDAF}}. The following functionalities are as defined herein:
+
+| Functionality       | Type          | Definition                  |
+|:--------------------|:--------------|:----------------------------|
+| `XofFixedKeyAes128` | Constructor   | {{Section 6.2 of !VDAF}}    |
+| `XofTurboShake128`  | Constructor   | {{Section 6.2 of !VDAF}}    |
+| `xof.next`          | Class Method  | {{Section 6.2 of !VDAF}}    |
+{: #XOF-functionalities title="XOF Functionalities from {{!VDAF}}."}
+
+> NOTE `xof` instances use a seed of size `SEED_SIZE` (in bytes).
+
+Each invocation of a XOF is initialized with a domain separation tag set to
+`b'mastic' + to_le_bytes(VERSION, 4) + byte(usage) + ctx`, where `ctx` is the
```

Comment:
This has been updated:
https://github.com/jimouris/draft-mouris-cfrg-mastic/blob/fc295a793362793eebe9a6132a2b5881fb51fd91/poc/dst.py#L30-L42

Reply:
Fixed based on dst and dst_alg